erpnext/accounts/doctype/purchase_invoice/purchase_invoice.js

```diff
@@ -365,8 +365,12 @@ function hide_fields(doc) {
 	var parent_fields = ['due_date', 'is_opening', 'advances_section', 'from_date', 'to_date'];
 
 	if(cint(doc.is_paid) == 1) {
+		cur_frm.set_df_property('mode_of_payment', 'reqd', 1)
```

Comment:
You can bring this out of the condition and use the helper function:

```suggestion
		cur_frm.toggle_reqd('mode_of_payment', cint(doc.is_paid) === 1)
```